doctr/datasets/loaders/detection/core.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (C) 2021, Mindee.
+
+# This program is licensed under the Apache License version 2.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import os
+import json
+import cv2
+import tensorflow as tf
+import numpy as np
+from typing import List, Tuple
+
+
+def load_annotation(
+    labels_path: str,
+    img_name: str,
+) -> Tuple[List[List[List[int]]], List[List[List[int]]]]:
+    """Loads detection annotations (boxes) for an image, from a folder containing
+    annotations in json files for each image
+
+    Agrs:
+        labels_path: path to the folder containing all json annotations
+        img_name: name of the image to find the corresponding json annotation
+
+    Returns:
+        A tuple of 2 lists: a list of polygons to draw on the segmentation gt map,
+        and a list of ambiguous polygons to mask during the training
+
+    """
+    with open(os.path.join(labels_path, img_name + '.json')) as f:
+        labels = json.load(f)
+
+    polys = [[[int(x), int(y)] for [x, y] in polygon] for polygon in labels["boxes_1"] + labels["boxes_2"]]
+    polys_mask = [[[int(x), int(y)] for [x, y] in polygon] for polygon in labels["boxes_3"]]
+
+    return polys, polys_mask
+
+
+def resize(image, input_size):
+    return cv2.resize(image, input_size, interpolation=cv2.INTER_AREA)
+
+
+class DataGenerator(tf.keras.utils.Sequence):
+    """Data loader for Differentiable Binarization detection model
+
+    Args:
+        input_size: size (h, w) for the images
+        images_path: path to the images folder
+        labels_path: pathe to the folder containing json label for each image
+        batch_size: batch size to train on
+        suffle: if True, dataset is shuffled between each epoch
+
+    """
+    def __init__(
+        self,
+        input_size: Tuple[int, int],
+        images_path: str,
+        labels_path: str,
+        batch_size: int = 1,
+        shuffle: bool = True,
+    ) -> None:
+        self.input_size = input_size
+        self.batch_size = batch_size
+        self.images_path = images_path
+        self.labels_path = labels_path
+        self.shuffle = shuffle
+        self.on_epoch_end()
+        self.min_size_box = 3
+
+    def __len__(self):
+        # Denotes the number of batches per epoch
+        return int(np.floor(len(os.listdir(self.images_path)) / self.batch_size))
```

Comment:
Are we positive that when it's not a multiple of the batch size, it does not create one last unfull batch? (if so `math.ceil` might be useful)